src/node.cc

```diff
@@ -2087,12 +2087,12 @@ void DLOpen(const FunctionCallbackInfo<Value>& args) {
     return;
   }
   if (mp->nm_version != NODE_MODULE_VERSION) {
-    uv_dlclose(&lib);
     char errmsg[1024];
     snprintf(errmsg,
              sizeof(errmsg),
              "Module version mismatch. Expected %d, got %d.",
              NODE_MODULE_VERSION, mp->nm_version);
+    uv_dlclose(&lib);
```

Comment:
`mp->nm_version` is a pointer to memory from the shared object?  Can you add a comment explaining that?  It's not very obvious from just looking at the code.